include/graph_configuration.h

```diff
@@ -0,0 +1,93 @@
+
+#include <iostream>
+static constexpr char config_loc[] = "streaming.conf";
+
+// TODO: Replace this with an enum defined by GutterTree repo
+enum GutterSystem {
+  GUTTERTREE,
+  STANDALONE,
+  CACHETREE
+};
+
+// Graph parameters
+class GraphConfiguration {
+public:
+  // which guttering system to use for buffering updates
+  GutterSystem gutter_sys = STANDALONE;
+
+  // Where to place on-disk datastructures
+  std::string disk_dir = ".";
+
+  // Backup supernodes in memory or on disk when performing queries
+  bool backup_in_mem = true;
+
+  // The number of graph workers
+  size_t num_groups = 1;
+
+  // How many OMP threads each graph worker uses
+  size_t group_size = 1;
+
+  void read_configuration() {
```

Comment:
Instead of making the fields public and having a read option, can we just create two constructors; one where you provide all the values and one where you provide a file location?